### PR TITLE
test: fix test for cdcv2

### PR DESCRIPTION
**Pull request by ruchirK**
Previously, this test tried to emulate a produce belatedly sending a stream of
data that CDCv2 should ignore, but it did not send exact copies of ((k, v) t, d)
as expected and instead forwarded the times to a later time. The test
also had two instances of the same ((k, v), t) in a batch which is not valid
CDCv2 input.

This commit fixes the test to emit actually previously emitted data that obeys
the requirement that CDCv2 batches are consolidated.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by ruchirK** [comment]
I extracted this out of #10141 because I think it makes sense to get this test fix in independently of the rest of that change.

**Comment by umanwizard** [comment]
@aljoscha added this test. He should comment on what it is supposed to mean and whether it's okay to change it.

File: test/testdrive/avro-cdcv2.td [hunk 1]
```diff
@@ -123,12 +123,12 @@ $ kafka-ingest format=avro topic=data schema=${schema} timestamp=3
 # that restarted and emits previously emitted data at a compacted timestamp
 
 $ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
-{"array":[{"data":{"id":5,"price":{"int":10}},"time":9,"diff":1}]}
-{"array":[{"data":{"id":5,"price":{"int":12}},"time":9,"diff":1}]}
-{"array":[{"data":{"id":5,"price":{"int":12}},"time":9,"diff":-1}]}
+{"array":[{"data":{"id":5,"price":{"int":10}},"time":5,"diff":1}]}
```
Comment (aljoscha) [inline test/testdrive/avro-cdcv2.td:126]: To preserve the spirit of the test, we could emit these at times `9`, `8`, and `9`, and adapt the progress statement below to still be from `0` to `10`. The idea was to test the scenario where a producer re-produces data, but at a compacted timestamp. And CDCv2 should filter these out succesfully.

**Review by aljoscha — CHANGES_REQUESTED** [review]

File: test/testdrive/avro-cdcv2.td (hunk 1 above)
Comment (ruchirK) [reply test/testdrive/avro-cdcv2.td:126]: @aljoscha it seems like we resolved this on slack but lmk if you still need me to make changes?

**Review by aljoscha — APPROVED** [review]

**Pull request merged**